text/cognito-construct-library.md

```diff
@@ -0,0 +1,68 @@
+---
+feature name: cognito-construct-library
+start date: 27/01/2020
+rfc pr:
+related issue: n/a
+---
+
+# Summary
+
+This RFC covers the design of Cognito's CDK construct library coverage.
+
+# Motivation
+
+The CDK constructs that are currently available in the CDK have a decent level
+of usage among customers. However, the current usage is via the 'Cfn'
+constructs and some basic features available via the `UserPool` construct. The
+current implementation of the `UserPool` construct only covers sign in type,
+user pool attributes and triggers.
+
+The goal of this RFC is to review the current `UserPool` construct for
+ergonomics in usability and extensibility, and to extend the features covered
+by the Cognito module.
+
+# Design Summary
+
+This RFC is structured as a working backwards document. Since the focus of this
+RFC is to propose the API design, the best way to propose this is to write up
+the future user guide of the Cognito module, as if it was complete.
+
+The bulk of the RFC is in the supporting document -
+[working-backwards-readme.md](./cognito-construct-library/working-backwards-readme.md)
```

Comment:
What's the benefit of managing this in a separate document?

Reply:
I can keep the RFC structure in this doc while keeping the readme in the format/language that would feature when it is actually shipped. Mixing them up could complicate how the sections are structured and the voice & tense of the language. This main document would still contain things around motivations, limitations, adoptions, breaking changes, roll out (if applicable), etc., while the working-backwards-readme doc can continue to focus purely on the user experience.

Further, I could break up the one working-backwards-readme I have now into 2 or several if they get too big. Separate docs for user pool and identity pool, for instance.